lib/ret_web/controllers/api/v1/avatar_controller.ex

```diff
@@ -2,45 +2,32 @@ defmodule RetWeb.Api.V1.AvatarController do
   use RetWeb, :controller
 
   alias Ret.{Account, Repo, Avatar, AvatarListing, Storage, GLTFUtils}
-  import Ecto.Query
 
   plug(RetWeb.Plugs.RateLimit when action in [:create, :update])
 
-  defp get_avatar(avatar_sid) do
+  defp get_avatar(avatar_sid, preloads \\ []) do
     avatar_sid
     |> Avatar.avatar_or_avatar_listing_by_sid()
-    |> preload()
+    |> preload(preloads)
   end
 
-  defp preload(nil = _avatar) do
-    nil
+  defp preload(a) do
```

Comment:
do you still need to cover the `nil` case?

Reply:
Should be handled by the fallthrough to
```
  defp preload(_avatar, _preloads), do: nil
```